WooCommerce/Classes/POS/Presentation/Reusable Views/POSModalManager.swift

```diff
@@ -1,15 +1,26 @@
 import SwiftUI
+import Combine
 
 class POSModalManager: ObservableObject {
     @Published private(set) var isPresented: Bool = false
     @Published private(set) var allowsInteractiveDismissal: Bool = true
     private var contentBuilder: (() -> AnyView)?
     private var onDismiss: (() -> Void)?
+    private var subscriptions = Set<AnyCancellable>()
 
     func present<Content: View>(onDismiss: @escaping () -> Void, content: @escaping () -> Content) {
         contentBuilder = { AnyView(content()) }
         self.onDismiss = onDismiss
         isPresented = true
+
+        $isPresented
+            .sink { [weak self] _ in
+                guard let self else { return }
+                if !isPresented {
+                    reset()
```

Comment:
Are you able to land inside this statement? I can hit the `!isPresented` breakpoint every time I present the modal, but when is read, it's always `true`, no matter if we're presenting it or dismissing it either tapping outside the modal or in the "x".  So `reset()` might not being called? 

https://github.com/user-attachments/assets/f373934a-d872-4289-b798-9ae52ab1d225

Reply:
This is not meant to be called in normal circumstances. I only added it as an extra precaution in case `reset()` is not called. But it's likely unnecessary. I can remove it.